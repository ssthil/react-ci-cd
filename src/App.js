import React, { Component } from 'react'
import {
  Container,
  Grid,
  Column,
  Header,
  Card, Icon, Image
} from 'semantic-ui-react'

import { getImageUrl, getMovieInfo } from '../src/api'

//const API_KEY = `cfe422613b250f702980a3bbf9e90716`
//const API_URL = `https://api.themoviedb.org/3/movie/popular?&api_key=${API_KEY}`

const style = {
  fluid: {
    padding: '1em',
  },
  header: {
    margin: '1em 0em ',
  },
  h3: {
    marginTop: '2em',
    padding: '2em 0em',
  },
  last: {
    marginBottom: '300px',
  },
}

export default class App extends Component {

  constructor(props) {
    super(props)

    this.state = {
      moviesInfo: []
    }

  }

  componentDidMount() {
    getMovieInfo().then((data) => {
      this.setState({
        moviesInfo: data.results
      })
    })
  }

  render() {
    console.log(this.state.moviesInfo)
    return (
      <Container fluid style={style.fluid}>
        <Header as='h2' style={style.header}>The Popular Movie Bank</Header>
        <p>Popular Movies</p>
        <Grid relaxed='very' columns={5}>

          {
            this.state.moviesInfo.map((movie, i) => {
              return (
                <Grid.Column key={i}>
                  <Card>
                    <Image src={getImageUrl(movie.poster_path)} alt={movie.title} />
                    <Card.Content>
                      <Card.Header>
                        {movie.title}
                      </Card.Header>
                      <Card.Meta>
                        <span className='date'>
                        {movie.release_date}
                        </span>
                      </Card.Meta>
                      <Card.Description>
                        {movie.vote_average}
                      </Card.Description>
                    </Card.Content>
                  </Card>
                </Grid.Column>
              )
            })
          }

        </Grid>
      </Container>
    )
  }
}