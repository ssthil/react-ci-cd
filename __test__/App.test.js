import React from 'react'
import renderer from 'react-test-renderer'

import App from '../src/App';

describe('Welcome (Snapshot)', () => {
  it('renders without crashing', () => {
    const component = renderer.create(<App />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
})