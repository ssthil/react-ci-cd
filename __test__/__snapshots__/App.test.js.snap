// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome (Snapshot) renders without crashing 1`] = `
<div
  style={
    Object {
      "padding": "1em",
    }
  }
>
  <h2
    as="h2"
    style={
      Object {
        "margin": "1em 0em ",
      }
    }
  >
    The Popular Movie Bank
  </h2>
  <p>
    Popular Movies
  </p>
  <div>
    <div>
      <div>
        <div>
          <h3>
            Deadpool 2
          </h3>
          <h4>
            <span
              className="date"
            >
              2018-05-15
            </span>
          </h4>
          <p>
            7.9
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Avengers: Infinity War
          </h3>
          <h4>
            <span
              className="date"
            >
              2018-04-25
            </span>
          </h4>
          <p>
            8.5
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Solo: A Star Wars Story
          </h3>
          <h4>
            <span
              className="date"
            >
              2018-05-23
            </span>
          </h4>
          <p>
            6.8
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Thor: Ragnarok
          </h3>
          <h4>
            <span
              className="date"
            >
              2017-10-25
            </span>
          </h4>
          <p>
            7.4
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Fifty Shades Freed
          </h3>
          <h4>
            <span
              className="date"
            >
              2018-01-17
            </span>
          </h4>
          <p>
            6
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Jurassic World: Fallen Kingdom
          </h3>
          <h4>
            <span
              className="date"
            >
              2018-06-06
            </span>
          </h4>
          <p>
            7.8
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Black Panther
          </h3>
          <h4>
            <span
              className="date"
            >
              2018-02-13
            </span>
          </h4>
          <p>
            7.3
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Annihilation
          </h3>
          <h4>
            <span
              className="date"
            >
              2018-02-22
            </span>
          </h4>
          <p>
            6.3
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Wonder Woman
          </h3>
          <h4>
            <span
              className="date"
            >
              2017-05-30
            </span>
          </h4>
          <p>
            7.2
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Jurassic World
          </h3>
          <h4>
            <span
              className="date"
            >
              2015-06-06
            </span>
          </h4>
          <p>
            6.5
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Guardians of the Galaxy
          </h3>
          <h4>
            <span
              className="date"
            >
              2014-07-30
            </span>
          </h4>
          <p>
            7.9
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Red Sparrow
          </h3>
          <h4>
            <span
              className="date"
            >
              2018-03-01
            </span>
          </h4>
          <p>
            6.4
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Star Wars: The Last Jedi
          </h3>
          <h4>
            <span
              className="date"
            >
              2017-12-13
            </span>
          </h4>
          <p>
            7.1
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Aage Ki Soch
          </h3>
          <h4>
            <span
              className="date"
            >
              1988-01-01
            </span>
          </h4>
          <p>
            0
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            The Maze Runner
          </h3>
          <h4>
            <span
              className="date"
            >
              2014-09-10
            </span>
          </h4>
          <p>
            7
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            A Quiet Place
          </h3>
          <h4>
            <span
              className="date"
            >
              2018-04-05
            </span>
          </h4>
          <p>
            7.3
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Death Wish
          </h3>
          <h4>
            <span
              className="date"
            >
              2018-03-02
            </span>
          </h4>
          <p>
            5.5
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Meet Me In St. Gallen
          </h3>
          <h4>
            <span
              className="date"
            >
              2018-02-07
            </span>
          </h4>
          <p>
            5
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            War for the Planet of the Apes
          </h3>
          <h4>
            <span
              className="date"
            >
              2017-07-11
            </span>
          </h4>
          <p>
            6.9
          </p>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <h3>
            Thor
          </h3>
          <h4>
            <span
              className="date"
            >
              2011-04-21
            </span>
          </h4>
          <p>
            6.6
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
